src/app/tasks/task.service.ts

```diff
@@ -3,13 +3,14 @@ import { IdGeneratorService } from "./id-generator-service";
 import { Injectable } from "@angular/core";
 import { HttpClient } from "@angular/common/http";
 import { beApiRoutes } from "../routes/be-api.routes";
+import { BehaviorSubject } from 'rxjs';
 
 @Injectable()
 export class TaskService {
     
-    private taskTypes: Option[] = [];
-    private taskStatuses: Option[] = [];
-    private tasks: Task[] = [];    
+    private taskTypesSubject = new BehaviorSubject<Option[]>([] as Option[]);
```

Comment:
You don't have to specify the type of array in the default array, since it'll jsut be an empty array